workflow/rules/annotate.smk

```diff
@@ -1,101 +1,155 @@
-        candidates_with_annotation="results/calling/candidates/{sample}.sorted.bcf",
-        candidates_with_annotation_index="results/calling/candidates/{sample}.sorted.bcf.csi",
-        header_lines="results/calling/annotation/{sample}.header_lines.txt",
+        bcf="results/calling/calls/filtered_fdr/{group}.bcf",
+        sorted_header="results/calling/calls/filtered_fdr/reheader/{group}.header.sorted.txt",
     output:
-        variants="results/calling/calls/annotated/{sample}.bcf",
-        annotation=temp("results/calling/candidates/{sample}.sorted.bcf.tab"),
-        annotation_bgzip=temp("results/calling/candidates/{sample}.sorted.bcf.tab.bgz"),
-        annotation_bgzip_tabix=temp(
-            "results/calling/candidates/{sample}.sorted.bcf.tab.bgz.tbi"
-        ),
+        bcf="results/calling/calls/filtered_fdr/reheader/{group}.bcf",
     log:
-        "logs/re-annotate/{sample}.log",
-    benchmark:
-        "benchmarks/re-annotate/{sample}.txt"
+        "logs/reheader_filtered_bcf/{group}.log",
     conda:
-        "../envs/vcf_annotate.yaml"
-    params:
-        header=copy_annotation_vembrane_header_expr(),
-        table_expr=copy_annotation_table_expr(),
-        columns=copy_annotation_bcftools_annotate_columns(),
+        "../envs/bcftools.yaml"
     shell:
+        ## bcftools re-header seems to re-order entries
+        # bcftools reheader --header {input.sorted_header} --output {output.bcf} {input.bcf}
+        ## so we have to re-header ourselves
         """
-        vembrane table --header {params.header:q} {params.table_expr:q} {input.candidates_with_annotation} > {output.annotation} 2> {log}
-        bgzip -c {output.annotation} > {output.annotation_bgzip} 2>> {log}
-        tabix -p vcf --zero-based -S 1 -f {output.annotation_bgzip} 2>> {log}
-        bcftools annotate --header-lines {input.header_lines} --annotations {output.annotation_bgzip} --columns {params.columns} --output-type b --output {output.variants} {input.variants}  2>> {log}
+        cat {input.sorted_header} <(bcftools view -H {input.bcf}) | bcftools view -Ob > {output.bcf} 2> {log}
         """
 
 
-rule annotate_genes:
+rule sort_bcf_header:
     input:
-        annotation="resources/gencode.v38.annotation.sorted.gff3.gz",
-        variants="results/calling/calls/merged/{sample}.bcf",
+        bcf="results/calling/calls/filtered_fdr/{group}.bcf",
+        header="results/calling/calls/filtered_fdr/{group}.header.txt",
     output:
-        variants="results/calling/calls/pre_annotated/{sample}.bcf",
-    threads: 1
-    log:
-        "logs/annotate_genes/{sample}.log",
-    benchmark:
-        "benchmarks/annotate_genes/{sample}.txt"
+        sorted_header="results/calling/calls/filtered_fdr/reheader/{group}.header.sorted.txt",
     conda:
         "../envs/gff.yaml"
+    log:
+        "logs/sort_bcf_header/{group}.log",
     script:
-        "../scripts/gff_annotate.py"
+        "../scripts/sort_bcf_header.py"
```

Comment:
Correct, historic baggage, if you will ;)
I'm not a fan of this script anyways, as it makes some assumptions which may not always hold. But it was necessary because at some point in the pipeline (during bcftools re-header I think), the header gets jumbled such that the file becomes invalid.

Reply:
OK, I removed the `gff.yaml` and the reference here, for now. We can always replace / improve scripts...